pkg/rest/server.go

```diff
@@ -239,3 +358,150 @@ func sendError(w http.ResponseWriter, message string, err error) {
 		Message: fmt.Sprintf("%s: %v", message, err),
 	})
 }
+
+// MessagesLastMinute holds the count of produced and consumed messages in the last minute and last second
+type MessagesLastMinute struct {
+	Topic          string `json:"topic"`
+	ProducedCount  int64  `json:"produced_count"`
+	ConsumedCount  int64  `json:"consumed_count"`
+	ProducedPerSec int64  `json:"produced_per_sec"`
+	ConsumedPerSec int64  `json:"consumed_per_sec"`
+}
+
+// OffsetHistoryEntry stores offset and timestamp
+type OffsetHistoryEntry struct {
+	Offset    int64
+	Timestamp time.Time
+}
+
+// In-memory rolling window for offsets (not production safe)
+var producedHistory = make(map[string][]OffsetHistoryEntry)
+var consumedHistory = make(map[string][]OffsetHistoryEntry)
+
+// In-memory store for last offsets and timestamps (for demo purposes, not production safe)
+
+// Handler for messages per minute
+func (s *Server) handleMessagesPerMinute(w http.ResponseWriter, r *http.Request) {
+	admin, err := sarama.NewClusterAdminFromClient(s.kafkaClient)
+
+	if err != nil {
+		sendError(w, "Failed to create admin client", err)

```

Comment:
in previous http handlers you were building the response manually with http.Error. Can you use this `sendError` function everywhere for consistency?